### PR TITLE
Add XML hashcode to PDF

### PR DESCRIPTION
**Pull request by rnijveld**
- ~~Note: #685 should probably be merged first~~
- Adds XML Hashcode to the PDF
- Changes indentation of XML to 2 spaces
- ~~Example output: [model-na-31-2.pdf](https://github.com/user-attachments/files/18021786/model-na-31-2.pdf)~~
- Example output: [model-na-31-2-22.pdf](https://github.com/user-attachments/files/18024332/model-na-31-2-22.pdf)

### CODE REVIEW

File: backend/src/eml/base.rs [hunk 1]
```diff
@@ -79,13 +79,38 @@ pub trait EMLDocument: Sized + DeserializeOwned + Serialize {
...
                 .expect("Constructing a new XML Serializer cannot fail");
-            s.indent(' ', 4);
+            s.indent(' ', 2);
             s
         };
         Serialize::serialize(&self, serializer)
     }
 }
 
+pub fn eml_document_hash(input: &str, chunked: bool) -> String {
```
Comment (praseodym) [inline backend/src/eml/base.rs:89]: I think it's good to test this function, e.g.
```rust
    #[test]
    fn test_eml_document_hash() {
        let input = "test";
        let hash = eml_document_hash(input, false);
        assert_eq!(
            hash,
            "9F86D081884C7D659A2FEAA0C55AD015A3BF4F1B2B0B822CD15D6C15B0F00A08"
        );
        let hash = eml_document_hash(input, true);
        assert_eq!(
            hash,
            "9F86 D081 884C 7D65 9A2F EAA0 C55A D015 A3BF 4F1B 2B0B 822C D15D 6C15 B0F0 0A08"
        );
    }
```

File: backend/Cargo.toml [hunk 2]
```diff
@@ -37,6 +37,8 @@ tracing = "0.1.41"
 typst = "0.12.0"
 typst-pdf = "0.12.0"
 quick-xml = { version = "0.37.1", features = ["serialize"] }
+sha2 = "0.10.8"
+const-hex = "1.14.0"
```
Comment (praseodym) [inline backend/Cargo.toml:41]: Can you elaborate why you chose `const-hex`? It looks like it relies on some [unsafe Rust](https://github.com/search?q=repo%3ADaniPopes%2Fconst-hex%20unsafe&type=code) for performance, but we don't need it to be fast. 

The [hex crate](https://crates.io/crates/hex) is already in our dependency tree and it doesn't look like that one uses unsafe Rust.

File: backend/src/eml/base.rs (hunk 1 above)
Comment (rnijveld) [reply backend/src/eml/base.rs:89]: I copied your suggestion into a1ab3de98bb3c9e900a1e9af1e0ed65bb74a4f33 :)

File: backend/Cargo.toml (hunk 2 above)
Comment (rnijveld) [reply backend/Cargo.toml:41]: My main reason for not using hex is that it is no longer maintained and appears to be as good as abandoned. Alternatively there is also [faster-hex](https://crates.io/crates/faster-hex), but that one also has some unsafe usages.

Comment (rnijveld) [reply backend/Cargo.toml:41]: Hmm I just noticed some new commits on the repository, maybe it will come back to life again?

Comment (rnijveld) [reply backend/Cargo.toml]: As discussed on mattermost: switched to `hex` in cf5ada3fd988d9e4a4b5e0e57d9cb0f316752845

File: backend/templates/inputs/model-na-31-2.json [hunk 3]
```diff
@@ -640,6 +640,6 @@
       "locality": "Heemdamseburg"
     }
   ],
-  "hash": "ED36 60EB 017A 0D3A D3EF 72B1 6865 F991 A36A 9F92 72D9 1516 39CD 422B 4756 D161",
-  "creation_date_time": "04-12-2024 12:08:55"
+  "hash": "ed36 60eb 017a 0d3a d3ef 72b1 6865 f991 a36a 9f92 72d9 1516 39cd 422b 4756 d161",
+  "creation_date_time": "04-12-2024 12:08"
```
Comment (praseodym) [inline backend/templates/inputs/model-na-31-2.json:644]: Why the change to lowercase and no seconds in the timestamp?

Comment (jorisleker) [reply backend/templates/inputs/model-na-31-2.json:644]: Lowercase on my request for better readability (and to save some horizontal space)

Comment (rnijveld) [reply backend/templates/inputs/model-na-31-2.json:644]: With seconds in the timestamp the footer format is very cramped, especially as the number of pages goes over 100. The full timestamp (up to the nanosecond if we want) is always in the EML anyway.

**Review by praseodym — APPROVED** [review]

**Review by marlonbaeten — DISMISSED** [review]
Maybe we can remove the hex dependency since Rust already has format methods for this - see for instance: https://github.com/tweedegolf/etes/blob/e5b583ac122d0d2221c3e0aa31a1ba9b49b1f974/src/util.rs#L37

**Review by praseodym — DISMISSED** [review]

**Review by praseodym — APPROVED** [review]

**Review by oliver3 — APPROVED** [review]
